src/js/playground/playground.js

```diff
@@ -222,32 +222,63 @@ export class Playground {
     output.value = this.Blockly.Xml.domToPrettyText(xml);
     output.focus();
     output.select();
-    // this.taChange();
   }
-  fromXml() {
-    const input = document.getElementById('importExport');
-    const xml = this.Blockly.Xml.textToDom(input.value);
-    this.Blockly.Xml.domToWorkspace(xml, this.workspace);
-    // this.taChange();
+  fromJSON() {
+    try {
+      const input = document.getElementById('importExport');
+      const json = JSON.parse(input.value);
+      if (json.scriptList !== undefined && json.scriptList.length > 0) {
+        for (let i = 0; i < json.scriptList.length; i++) {
+          jsonDomToWorkspace(json.scriptList[i], this.workspace);
+        }
+      }
+      else {
+        jsonDomToWorkspace(json, this.workspace);
+      }
+    } catch (e) {
+      console.error(e.message);
+      console.error("Can't render current JSON object - object has to start with scriptList.");
+    }
   }
   fromParser() {
-    const input = document.getElementById('importExport');
-    const blocksXml = this.Parser.convertScriptString(input.value);
-    console.log(blocksXml);
-
-    if (blocksXml === undefined || blocksXml === "") {
-      throw "no response from XStreamParser";
-    } else {
-      this.Blockly.Xml.domToWorkspace(wrapElement(blocksXml.firstChild, 'xml'), this.workspace);
+    const beforeScript = `<?xml version="1.0" encoding="UTF-8" standalone="yes"?><program><header>
```

Comment:
`fromParser()` renders blocks with `convertProgramToJSONDebug()` and new `jsonDomToWorkspace()` function.